kernel/idle.c

```diff
@@ -41,6 +55,14 @@ void idle(void *unused1, void *unused2, void *unused3)
 	__ASSERT_NO_MSG(_current->base.prio >= 0);
 
 	while (true) {
+#ifdef CONFIG_SHUTDOWN
+		if ((atomic_get(&shutdown_sheduled) == 1) &&
+		    (atomic_get(&_cpus_active) == 1)) {
```

Comment:
you are not preventing other cores to wake up, so you can definitely have a race condition where another core can wake up and start things while you are shutting down or you never  get this condition true.

Reply:
or I missed something here :)